internal/log/klog.go

```diff
@@ -67,7 +67,7 @@ func writeKeysAndValues(b *strings.Builder, keysAndValues ...any) {
 			}
 		}
 
-		if i == len(keysAndValues) {
+		if i+2 == len(keysAndValues) {
```

Comment:
hm it's not immediately clear to me what this piece is doing

Reply:
Looks like a bracket index matching fix. But... Should probably be a separate Pull Request indeed.

Reply:
The last index `i` of this loop should be `i = len(keysAndValues) - 2` because `i += 2` in each loop, and `i` never equals to `len(keysAndValues)`.
Since this is a trivial change and related to klog, I guess it's ok to have it in this PR.

Reply:
actually never mind, I didn't take into account when `keysAndValues` is odd number, which shouldn't happen though.
I'll remove this fix for now.